packages/astro/test/experimental-content-collections-cache-invalidation.test.js

```diff
@@ -0,0 +1,98 @@
+import assert from 'node:assert/strict';
+import { after, before, describe, it } from 'node:test';
+import { loadFixture } from './test-utils.js';
+import fs from 'node:fs';
+import { copyFiles } from '../dist/core/build/static-build.js';
+
+describe('Experimental Content Collections cache - invalidation', () => {
+	class CacheBackup {
+		constructor(root, relCacheDir) {
+			this.root = new URL(root, import.meta.url);
+			this.cacheDir = new URL(relCacheDir, this.root);
+			this.tmpDir = new URL(`./tmp` + relCacheDir.slice(1), this.root);
+		}
+		backup() {
+			this.rmTmp();
+			copyFiles(this.cacheDir, this.tmpDir);
+		}
+		restore() {
+			fs.rmSync(this.cacheDir, { recursive: true });
+			copyFiles(this.tmpDir, this.cacheDir);
+		}
+		rmTmp() {
+			fs.rmSync(this.tmpDir, { force: true, recursive: true });
+		}
+	}
+
+	class ManifestUsed {
```

Comment:
nit: `TestPlugin`? It took me a while to understand that

Reply:
sure